draft-ietf-ppm-dap.md

```diff
@@ -1291,44 +1295,70 @@ struct {
 } HpkeConfig;
 
 opaque HpkePublicKey<0..2^16-1>;
-uint16 HpkeAeadId; /* Defined in [HPKE] */
-uint16 HpkeKemId;  /* Defined in [HPKE] */
-uint16 HpkeKdfId;  /* Defined in [HPKE] */
+uint16 HpkeAeadId;
+uint16 HpkeKemId;
+uint16 HpkeKdfId;
 ~~~
 
-Aggregators MUST allocate distinct id values for each `HpkeConfig` in an
+The possible values for `HpkeAeadId`, `HpkeKemId` and `HpkeKdfId` are as defined
+in {{!HPKE, Section 7}}.
+
+Aggregators MUST allocate distinct `id` values for each `HpkeConfig` in an
 `HpkeConfigList`.
 
-The Client MUST abort if any of the following happen for any HPKE config
-request:
+The Client MUST abort if:
 
-* the GET request did not return a valid HPKE config list;
+* the response is not a valid HPKE config list;
 * the HPKE config list is empty; or
-* no HPKE config advertised by the Aggregator specifies a supported a KEM, KDF,
-  or AEAD algorithm triple.
-
-Aggregators SHOULD use HTTP caching to permit client-side caching of this
-resource {{!RFC5861}}. Aggregators SHOULD favor long cache lifetimes to avoid
-frequent cache revalidation, e.g., on the order of days. Aggregators can control
-this cached lifetime with the Cache-Control header, as in this example:
-
-~~~ http-message
-  Cache-Control: max-age=86400
+* no HPKE config advertised by the Aggregator specifies a supported KEM, KDF and
+  AEAD algorithm triple.
+
+Aggregators SHOULD use caching to permit client-side caching of this resource
+{{!RFC9111}}. Aggregators SHOULD favor long cache lifetimes to avoid frequent
+cache revalidation, e.g., on the order of days. Aggregators can control this
+cached lifetime with the Cache-Control header.
+
+Servers should choose a Cache-Control value appropriate to the lifetime of their
+keys. Clients SHOULD follow the usual caching semantics {{!RFC9111}} for HPKE
+configurations.
+
+Long cache lifetimes may result in Clients using stale HPKE configurations.
+Aggregators SHOULD continue to accept reports with old keys for at least twice
+the cache lifetime in order to avoid rejecting reports.
+
+#### Example
+
+~~~ http
+GET /leader/hpke_config
+Host: example.com
+
+HTTP/1.1 200
+Content-Type: application/dap-hpke-config-list
+Cache-Control: max-age=86400
+
+encoded([
+  struct {
+    id = 194,
+    kem_id = 0x0010,
+    kdf_id = 0x0001,
+    aead_id = 0x0001,
+    public_key = [0x01, 0x02, 0x03, 0x04, ...],
+  } HpkeConfig,
+  struct {
+    id = 17,
+    kem_id = 0x0020,
+    kdf_id = 0x0001,
+    aead_id = 0x0003,
+    public_key = [0x04, 0x03, 0x02, 0x01, ...],
+  } HpkeConfig,
+])
```

Comment:
Add an example per https://datatracker.ietf.org/doc/html/rfc9205#section-4.1-5

I had to come up with a syntax for examples of TLS presentation language structs, with things like an implied `encoded()` function and the `[0x01, 0x02, ...]` syntax for byte arrays. Since this text is non-normative, I think this is good enough. For precedent, ACME doesn't exhaustively spell out what `base64url()` does in [its examples](https://datatracker.ietf.org/doc/html/rfc8555#section-7.3).

In the example, we don't say "200 OK". This is nitty, but 9205 says not to include the reason phrase: https://datatracker.ietf.org/doc/html/rfc9205#section-4.6-7